lib/SubjectByPropertyAnyValueHandler.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const { namespaceMap } = require( './rdfNamespaces' );
+const hasSingleTriple = require( './hasSingleTriple' );
+const CirrusSearchRequest = require( './CirrusSearchRequest' );
+
+module.exports = class SubjectByPropertyAnyValueHandler {
+	handle( _query, parsedQuery ) {
+		if ( !parsedQuery ||
+			parsedQuery.queryType !== 'SELECT' ||
+			parsedQuery.variables.length !== 1 ||
+			parsedQuery.variables[ 0 ].termType !== 'Variable' ||
+			!hasSingleTriple( parsedQuery ) ) {
+			return false;
+		}
+
+		const triple = parsedQuery.where[ 0 ].triples[ 0 ];
+		if ( triple.subject.termType !== 'Variable' ||
+			triple.subject.id !== parsedQuery.variables[ 0 ].id ||
+			triple.predicate.termType !== 'NamedNode' ||
+			!triple.predicate.id.startsWith( namespaceMap.Wikidata.wdt ) ||
+			triple.object.termType !== 'BlankNode' &&
+			(
+				triple.object.termType !== 'Variable' &&
+				!parsedQuery.variables.includes( triple.object.id )
+			) ) {
```

Comment:
This is making my head hurt :D but I think *in principle* it needs to be something like this instead:

```diff
-			triple.object.termType !== 'BlankNode' &&
-			(
-				triple.object.termType !== 'Variable' &&
-				!parsedQuery.variables.includes( triple.object.id )
-			) ) {
+			!(
+				triple.object.termType === 'BlankNode' ||
+				triple.object.termType === 'Variable' &&
+				!parsedQuery.variables.includes( triple.object.id )
+			) ) {
```

– except that the `includes` doesn’t work properly because `parsedQuery.variables` is a list of variable *objects*, not their string names. We could fix this with a `map`, but since we already know that there’s only a single variable selected, this is simpler:

```suggestion
			!(
				triple.object.termType === 'BlankNode' ||
				triple.object.termType === 'Variable' &&
				parsedQuery.variables[ 0 ].id !== triple.object.id
			) ) {
```

Reply:
Oh. Yes. Thanks, done! :D